### DISCUSSION TITLE
Ability to turn off ignoreclaims

### DISCUSSION THREAD

**Discussion by LogGits** [GriefPrevention/GriefPrevention#991] [Ideas]
**As** an op and/or staff member
**I want** to have a permission node to bypass claim restrictions instead of /ignoreclaims
**Because** it's tedious to have to type it every time I join and/or when a player asks to check which trustee griefed their house.

Additional info:
A permission to bypass claim restrictions would be alot easier and less tedious.

**Comment by fredagsguf** [comment]
i agree with this, i've personally added something that forces my staff to execute the command when they join.

Maybe instead of a permission, then make the toggle actually save when they log out - so its the same state when they join again?

**Comment by fredagsguf** [comment]
You probably can't use it as its using skript.
but its pretty simple, i just force staff to execute /ic when they join

**Comment by fredagsguf** [comment]
its literally 3 lines.. please learn how to skript urself.
`on join:`
`	if player has permission "your.own.permission":`
`		make player execute "/ic"`

**Comment by RoboMWM** [comment]
Ya, Big_Scary had an explicit reason for not implementing this, I'll see if I can find it.

**Comment by RoboMWM** [comment]
mk ya I should add this to the FAQ, he responded 3 separate times:

https://www.spigotmc.org/threads/griefprevention.35615/page-5#post-451552

> No, there's no option for that. You have to turn it on each time you log in. This is a safety feature - it's sometimes easy to not realize you're in someone's land claim when you are. By defaulting to off, you get a "warning" that tells you you're about to change someone's land claim. Then you can explicitly opt-into ignoring claims if that's what you really want to do. I think most admins actually like it this way - some have even asked me to automatically turn /ic off after some time to ensure they don't forget and leave it on.

https://www.spigotmc.org/threads/griefprevention.35615/page-82#post-1216689

> That's by design. GP won't let an admin accidentally change a claimed area. The way it does that is by insisting admins type /ic before they make changes where players have protected. If you want to risk shooting yourself in the foot by unintentionally wrecking someone's build and thus incurring the wrath of your players, then the cost for that will be four keystrokes per login.

https://www.spigotmc.org/threads/griefprevention.35615/page-88#post-1254461

> That feature might be broken, I plan to remove it anyway. The right way to do this is have admins use /ic to toggle claims override. It's only four keystrokes per play session, and prevents accidental oops-es that make players nuts and generate drama. Actually, I get many more requests to automatically turn off /ic after a time interval than I ever do for an option to make it always-on.

**Comment by RoboMWM** [comment]
That being said, feel free to use an addon or make a script (as mentioned here) to do this. (I just recalled that Big_Scary had an opinion on this.)

**Comment by fredagsguf** [comment]
> It's only four keystrokes per play session

its not only four keystrokes though, well alone it is - but theres multiple other plugins where you have to enable stuff etc - like /mcchatspy in mcmmo.

However i see the point, i just fell like theres better ways to do but i'm not smart enough to come up with them :)

**Comment by RoboMWM** [comment]
I'd see the use case if it were commonplace to have "admin-only" Minecraft accounts - but even then, being prompted when invoking admin privileges is still the norm anyways (UAC, `sudo`, etc.)

**Comment by fredagsguf** [comment]
Indeed, and lets keep it that way for the "Harming" commands.
but the /ic isn't that harmful in my opinion - staff wouldn't grief either way?

**Comment by RoboMWM** [comment]
Well it's more for accidental modification, not necessarily grief. I've done it before as well, I'd be in `/ic` when checking something else in another's claim, then I'd go back to playing/exploring and forgetting to turn off `/ic`, and after a while realizing that I totally shouldn't be able to break/modify stuff somewhere so ya.

**Comment by fredagsguf** [comment]
maybe a notification could be made, like when you have /ic enabled - it would tell you once in a while that you are currently breaking/modifying stuff inside a claim that doesn't belong to you (the staff member)
This doesn't mean that /ic needs to be changed like suggested above, this is kinda a standalone idea but usefull nonetheless :)

**Comment by bigpresh** [comment]
Just to add my point of view too, I'd never use an automatic ignoreclaims mode; I treat /ignoreclaims the same kind of way as using sudo / becoming root when managing a system - only do it with thought, and for as little as possible - I want to drop privileges as soon as possible.

Having to enable it when you want to use it leaves an audit log of when people are doing so, which can be useful.

Also, it protects me from accidental mis-clicks breaking other people's stuff (particularly if you happened to be in creative mode at the time), and makes it easier for me as a player to distinguish "I can break that/build there because it's unclaimed" (maybe an old abandoned build I've found, etc) to "I can break that/build there even though it's someone's claim because I have ignoreclaims active".  @Laukage's suggestion of a reminder notification is good, but would remind you *after* you'd made a mistake.

I wonder even if ignoreclaims should have an automatic timeout after which it is automatically disabled (the timer resetting each time you break a block that required your ignoreclaims to allow) so that you can't turn it on then accidentally leave it on?

**Comment by Jikoo** [comment]
While I don't support the idea of `/ic` as a permanent toggle, I think it would be great if it accepted a on/off parameter. That way all the users who do have other admin-mode plugins can create generic on/off commands using Bukkit's [commands.yml feature](https://bukkit.gamepedia.com/Commands.yml).
I.e.
```yaml
aliases:
  enablemod:
  - griefprevention:ignoreclaims on
  - myadminplugin:enablepower
  - otheradminplugin:sneaky true
  disablemod:
  - griefprevention:ignoreclaims off
  - myadminplugin:disablepower
  - otheradminplugin:sneaky false
```

In its current state this wouldn't work, because /disablemod would turn on ignore claims if it was off.

**Comment by bigpresh** [comment]
> I think it would be great if it accepted a on/off parameter.

Yeah, that would make sense - an optional param for desired state (on/off), and just toggle if no param.
